include/GooseEPM/System.h

```diff
@@ -334,6 +334,24 @@ class SystemAthermal {
         return m_propagator.shape();
     }
 
+    /**
+     * @brief Set the propagator.
```

Comment:
If the user is allowed to modify the propagator there should be some minimal documentation of the internal ordering. This was left out so far because the public method of required specifying the ordering. 

It is fine to have just a tiny example, see `detail::reorganise_popagator` for the internal organisation.

Reply:
Thinking about it again, maybe we should have the propagator as a read-only property? It makes sense to change fields like sigmay_mean after initialization (we could think of changing material properties, e.g. weakening) but IMO the propagator should be immutable. The only exception I see for now is if we want to change to a stress-driven protocol in the middle of a simulation. What do you think?

Reply:
I agree that it is better to make it read-only. If one really wants to change it one could also construct a new object. That can be done in such a way that it is not costly